libdirections-common/src/main/java/com/mapbox/navigation/route/common/NavigationRoute.kt

```diff
@@ -0,0 +1,782 @@
+package com.mapbox.navigation.route.common
+
+import android.content.Context
+import androidx.annotation.FloatRange
+import androidx.annotation.IntRange
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.MapboxDirections
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.core.exceptions.ServicesException
+import com.mapbox.core.utils.TextUtils
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.route.common.extension.getUnitTypeForLocale
+import com.mapbox.navigation.utils.extensions.inferDeviceLocale
+import okhttp3.EventListener
+import okhttp3.Interceptor
+import retrofit2.Call
+import retrofit2.Callback
+
+/**
+ * The NavigationRoute class wraps the [MapboxDirections] class with parameters which
+ * <u>must</u> be set in order for a navigation session to successfully begin. While it is possible
+ * to pass in any [com.mapbox.api.directions.v5.models.DirectionsRoute] into
+ * [MapboxNavigation.startNavigation], using this class will ensure your
+ * request includes all the proper information needed for the navigation session to begin.
+ *
+ * Developer Note: MapboxDirections cannot be directly extended since it is an AutoValue class.
+ *
+ * 0.5.0
+ */
+
+class NavigationRoute
```

Comment:
@andrlee made a good point in one of the previous comments - this class in the current form cannot be a part of the `common` directions library. If we are going to reuse this artifact in the `libdirections-onboard`, we're going to bring dependencies to retrofit and okhttp to a module that is not going to make any requests (at least from the platform code).

This whole class should either be moved to `libdirections-offboard` or at least the logic that makes uses networking libraries.

Reply:
lets discuss it in voice